src/components/RoomNameInput.js

```diff
@@ -26,6 +26,12 @@ const propTypes = {
     /** ID of policy whose room names we should be checking for duplicates */
     policyID: PropTypes.string,
 
+    /** Whether to show the error on change of input or not */
+    shouldShowErrorOnChange: PropTypes.bool,
```

Comment:
Not sure I understand where this is coming from. Why would we _not_ want to validate on change? We should aim to keep this simple because our form component currently in development will soon replace logic like this...

Reply:
I asked the question about this https://github.com/Expensify/App/pull/7098#issuecomment-102207487 to you.

Basically, Room SettingsPage shows the error on the change of the input. So to preserve that behavior and still utilize this component, this is the easiest way. 

If you say that we should also make that like other forms of the app, I would say that would be out of the scope of this issue.

Reply:
Not seeing what comment you're referring to - I think GH is having issues atm.

>I would say that would be out of the scope of this issue.

If we are adding things that don't need to be added then it's in scope 😄 

>Room SettingsPage shows the error on the change of the input

Where does this happen? Am I missing it? Can you give me an example screenshot or video? 

WRT form handling no errors should be shown on change unless the input is blurred first. I'm not sure it is worth preserving a behavior that will be changed very soon. Also, the entire feature is behind a beta?

Reply:
Regardless, it should be up to the parent I think. This component should be dumb and just show whatever error is passed to it.

Reply:
Yeah, validation should be handled via Parent but this component was designed to be reactive for https://github.com/Expensify/App/blob/9d6e07a0d3bb298d7edba16824efb95afc94f280/src/pages/ReportSettingsPage.js#L133-L154. 

To keep it compatible with that page I suggested using this prop.

But I showed my concern that the Settings page input also show error on submit like other forms.
![image](https://user-images.githubusercontent.com/24370807/152276871-7c1c2d7b-173b-4fff-984a-7e4e6360a0cb.png)

Reply:
What does "designed to be reactive" mean? Pretty sure errors should be passed to inputs and not have `onChangeError` callbacks. Seems like we maybe got the design wrong on this component.

@ABee-Tech can you please remove this prop for now?

Reply:
Okay, I also think onChangeError has no work as we are rendering error inside the component. I will remove `onChangeError`.

Reply:
@marcaaron The `onChangeError` is used for disabling Save button to check whether there is error or not on `ReportSettingsPage`. 

https://github.com/Expensify/App/blob/1d72a5e01ed6e0c0b500f7e0617d05eb8c05dee9/src/pages/ReportSettingsPage.js#L132-L155

So, I think we should change behavior of `ReportSettingsPage` also by changing validation from on change text to on form submit. This will cancel button disabling and change behavior as it should be.